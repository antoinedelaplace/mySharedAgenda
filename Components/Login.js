// Login.js

import React from 'react'
import { googleLogin } from '../Google/login/googleLogin'
import { StyleSheet, View } from 'react-native'
import { Button, Text } from 'react-native-elements'
import { connect } from 'react-redux'

class Login extends React.Component {

    constructor(props) {
        super(props);

        //Bind this
        this._login = this._login.bind(this);
    }

    _login() {
        googleLogin().then(data => {
            this.setState({
                user: data
            });
            const action = { type: "ADD_USER", value: this.state.user };
            this.props.dispatch(action);
            this.props.navigation.navigate(data ? 'MyCalendar' : 'Login');
        });
    }

    render() {
        return (
            <View style={styles.container} >
                <Text style={styles.mainTitle}>
                    Welcome to MySharedAgenda Application !
                </Text>
                <Button
                    icon={{
                        name: 'google',
                        type: 'font-awesome',
                        size: 18
                    }}
                    onPress={() => this._login()}
                    title="Login Google"
                    buttonStyle={{
                        justifyContent: "center",
                        backgroundColor: "#d34836",
                        width: 200,
                        height: 45,
                        borderColor: "transparent",
                        borderWidth: 0,
                        borderRadius: 20
                    }} />
            </View >
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: "#fff",
        alignItems: "center",
        justifyContent: "center"
    },
    mainTitle: {
        margin: 20
    }
});

const mapStateToProps = (state) => {
    return state
};

export default connect(mapStateToProps)(Login)