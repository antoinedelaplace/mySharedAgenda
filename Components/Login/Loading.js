// Loading.js

import React from 'react'
import { googleLogin } from '../../Google/login/googleLogin'
import { connect } from 'react-redux'
import { View, Text, ActivityIndicator, StyleSheet } from 'react-native'

class Loading extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            user: undefined
        };
    }

    componentDidMount() {
        googleLogin().then(data => {
            this.setState({
                user: data
            });
            const action = { type: "ADD_USER", value: this.state.user };
            this.props.dispatch(action);
            this.props.navigation.navigate(data ? 'MyCalendar' : 'Login');
        });
    }

    render() {
        return (
            <View style={styles.container}>
                <Text style={{ fontSize: 15, }}>it's coming... (Loading)</Text>
                <ActivityIndicator style={styles.loadingIcon} size='large' color='orange' />
            </View>
        )
    }
}

const mapStateToProps = (state) => {
    return state
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
    },
    loadingIcon: {
        marginTop: 20,
    }
});

export default connect(mapStateToProps)(Loading)