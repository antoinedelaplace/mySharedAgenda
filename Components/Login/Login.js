// Login.js

import React from 'react'
import { StyleSheet, View, Image } from 'react-native'
import { Button, Text } from 'react-native-elements'
import { googleLogin } from '../../Google/login/googleLogin'
import { connect } from 'react-redux'

class Login extends React.Component {

    constructor(props) {
        super(props);
        //Bind this
        this._login = this._login.bind(this);
    }

    _login() {
        googleLogin().then(data => {
            this.setState({
                user: data
            });
            const action = { type: "ADD_USER", value: this.state.user };
            this.props.dispatch(action);
            this.props.navigation.navigate(data ? 'MyCalendar' : 'Login');
        });
    }

    render() {
        return (
            <View style={styles.container} >
                <View style={styles.logoContainer}>
                    <Image style={styles.logo} source={require('../../Images/launchscreen.png')}></Image>
                    <Button
                        icon={{
                            name: 'google',
                            type: 'font-awesome',
                            size: 18
                        }}
                        onPress={() => this._login()}
                        title="Login Google"
                        buttonStyle={{
                            marginTop: 20,
                            backgroundColor: "#C97C00",
                            width: 170,
                            height: 45,
                            borderColor: "transparent",
                            borderWidth: 0,
                            borderRadius: 10
                        }} />
                </View>
                <View style={styles.footerContainer}>
                    <Text style={styles.footerContent}>An app made MySharedAgenda using React Native â’¸ Copyright AFY</Text>
                </View>
            </View >
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#f39c12',
    },
    logoContainer: {
        alignItems: 'center',
        flex: 1,
        justifyContent: 'center'
    },
    logo: {
        width: 150,
        height: 150,
    },
    footerContainer: {
        alignItems: 'center',
        justifyContent: 'center'
    },
    footerContent: {
        fontSize: 12,
        marginBottom: 20,
        color: '#fff',
        textAlign: 'center',
        width: 300,
        opacity: 0.85,
    },
});

const mapStateToProps = (state) => {
    return state
};

export default connect(mapStateToProps)(Login)