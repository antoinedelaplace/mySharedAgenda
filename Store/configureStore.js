import { combineReducers } from 'redux'
import { createStore } from 'redux'
import changeUser from './Reducers/userReducer'
import changeCalendar from './Reducers/calendarReducer'

let rootReducer = combineReducers({ userReducer: changeUser, calendarReducer: changeCalendar })

//Cr√©ation du store avec nos reducers
export default createStore(rootReducer);