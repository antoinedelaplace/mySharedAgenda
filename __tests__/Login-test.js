import React from 'react';
import Login from '../Components/Login'
import Store from '../Store/configureStore'

import renderer from 'react-test-renderer';

/* Login page renderer */
test('Login.js renders correctly', () => {
    const tree = renderer.create(<Login store={Store}/>).toJSON();
    expect(tree).toMatchSnapshot();
});