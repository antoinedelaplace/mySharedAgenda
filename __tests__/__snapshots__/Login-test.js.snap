// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login.js renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Array [
        Object {},
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        Object {
          "margin": 20,
        },
      ]
    }
  >
    Welcome to MySharedAgenda Application !
  </Text>
  <View
    style={
      Array [
        Object {
          "marginLeft": 15,
          "marginRight": 15,
        },
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        pointerEvents="box-only"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#9E9E9E",
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 19,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "padding": 12,
            },
            undefined,
            undefined,
            undefined,
            Object {
              "backgroundColor": "#d34836",
              "borderColor": "transparent",
              "borderRadius": 20,
              "borderWidth": 0,
              "height": 45,
              "justifyContent": "center",
              "width": 200,
            },
            undefined,
            undefined,
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 18,
              },
              Array [
                Object {
                  "marginRight": 10,
                },
                undefined,
              ],
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          type="font-awesome"
        >
          ï† 
        </Text>
        <Text
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontSize": 17.5,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ],
            ]
          }
        >
          Login Google
        </Text>
      </View>
    </View>
  </View>
</View>
`;
